File: bnim2219-spring/src/main/java/edu/bbte/idde/bnim2219/spring/model/dto/outgoing/NoIdChoreDTO.java
package edu.bbte.idde.bnim2219.spring.model.dto.outgoing;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class NoIdChoreDTO {
    private String title;
    private String description;
    private Date deadline;
    private Integer priorityLevel;
    private Boolean done;
}

File: src/controller/dto/outgoing/noIdChore.dto.ts
import { Transform } from 'class-transformer';

export class NoIdChoreDTO {
  title!: string;
  description!: string;
  @Transform(({ value }) => value.toISOString().split('T')[0])
  deadline!: Date;
  priorityLevel!: number;
  done!: boolean;
}
